indexclient/client.py

```diff
@@ -526,6 +532,25 @@ def delete(self):
                             params={"rev": self.rev})
         self._deleted = True
 
+    def get_url_metadata_by_type(self, request_prop: str, url_type: str) -> Optional[str]:
```

Comment:
I think having this method return the actual `URLMetadata` will more useful in the long run, you can then add explicit methods that return the state and the url named according the definitions in the ticket. It also makes the `URLMetadata` data structure much more useful, though I think a more appropriate definition of this data structure will be using `TypedDict`